sdk/src/test/java/com/mapbox/maps/ViewAnnotationManagerZOrderingTest.kt

```diff
@@ -0,0 +1,264 @@
+package com.mapbox.maps
+
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.FrameLayout
+import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.geojson.Point
+import com.mapbox.maps.ViewAnnotationManagerImpl.Companion.EXCEPTION_TEXT_GEOMETRY_IS_NULL
+import com.mapbox.maps.renderer.MapboxRenderThread
+import com.mapbox.maps.viewannotation.viewAnnotationOptions
+import io.mockk.*
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertThrows
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+internal class ViewAnnotationManagerZOrderingTest(
+  private val oldDescriptors: List<ViewAnnotationPositionDescriptor>,
+  private val newDescriptors: List<ViewAnnotationPositionDescriptor>,
+  private val shouldChangeZOrder: Boolean,
+) {
+
+  @Test
+  fun calculatesCorrectZOrder() {
+    assertEquals(
+      shouldChangeZOrder,
+      ViewAnnotationManagerImpl.needToReorderZ(
+        oldDescriptors, newDescriptors
+      ),
+    )
+  }
+
+  companion object {
+    private fun descriptors(vararg identifiers: String) = identifiers.map {
+      ViewAnnotationPositionDescriptor(
+        it, 0, 0, ScreenCoordinate(0.0, 0.0)
+      )
+    }
+
+    @JvmStatic
+    @Parameterized.Parameters
+    fun data() = listOf(
+      arrayOf(
+        /* old descriptors */
+        descriptors(
+          "1",
+          "2",
+          "3",
+        ),
+        /* new descriptors */
+        descriptors(
+          "1",
+          "2",
+        ),
+        /* change z order */
+        false,
+      ),
+      arrayOf(
+        /* old descriptors */
+        descriptors(
+          "1",
+          "2",
+          "3",
+        ),
+        /* new descriptors */
+        descriptors(
+          "1",
+          "3",
+          "2",
+        ),
+        /* change z order */
+        true,
+      ),
+      arrayOf(
+        /* old descriptors */
+        descriptors(
+          "1",
+          "2",
+          "3",
+        ),
+        /* new descriptors */
+        listOf<String>(),
+        /* change z order */
+        false,
+      ),
+      arrayOf(
+        /* old descriptors */
+        descriptors(),
+        /* new descriptors */
+        descriptors(
+          "1",
+          "3",
+          "2",
+        ),
+        /* change z order */
+        true,
+      ),
+      arrayOf(
+        /* old descriptors */
+        descriptors(
+          "1",
+          "3",
+        ),
+        /* new descriptors */
+        descriptors(
+          "1",
+          "2",
+        ),
+        /* change z order */
+        false,
```

Comment:
How come this is `false` but from empty to `1,3,2` is true (https://github.com/mapbox/mapbox-maps-android/pull/1744/files#diff-0e7802c1bcba87eaaea61f8bf7ba216fe00ab2f8eae240f5dfbcadd3429f9470R101) ?

Reply:
Hmm, true is from 1-2-3 to 1-3-2, if you mean this test case
<img width="387" alt="image" src="https://user-images.githubusercontent.com/2265924/195142302-21fdddd1-7c8d-4dd8-ab6c-3b2b63ffa6e6.png">

Reply:
Ah, also it's true for any update that is adding more elements that there were before. So empty to anything non-empty will produce true.

Reply:
1-3 to 1-2 does also assumes that new view will be placed into the layout and onMeasure will be called, we just do not optimize calculations this particular edge case.